homeassistant/components/n26/sensor.py

```diff
@@ -0,0 +1,288 @@
+        self._limits = limits
+        self._balance = balance
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True.
+
+        Data needs to be polled from the bank servers.
+        """
+        return True
+
+    def update(self) -> None:
+        """Get the current balance and currency for the account."""
+        self._account_info = self._client.get_account_info()
+        self._account_statuses = self._client.get_account_statuses()
+        self._limits = self._client.get_account_limits()
+        self._balance = self._client.get_balance()
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the entity."""
+        return self._balance["iban"]
+
+    @property
+    def name(self) -> str:
+        """Friendly name of the sensor."""
+        return "n26_%s" % self._balance["iban"].lower()
+
+    @property
+    def state(self) -> float:
+        """Return the balance of the account as state."""
+        return self._balance["bankBalance"]
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Use the currency as unit of measurement."""
+        return self._balance["currency"]
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Additional attributes of the sensor."""
+        attributes = {
+            ATTR_IBAN: self._balance.get("iban"),
+            ATTR_AVAILABLE_BALANCE: self._balance.get("availableBalance"),
+            ATTR_USABLE_BALANCE: self._balance.get("usableBalance"),
+            ATTR_ACC_OWNER_TITLE: self._account_info.get("title"),
+            ATTR_ACC_OWNER_FIRST_NAME: self._account_info.get("kycFirstName"),
+            ATTR_ACC_OWNER_LAST_NAME: self._account_info.get("kycLastName"),
+            ATTR_ACC_OWNER_GENDER: self._account_info.get("gender"),
+            ATTR_ACC_OWNER_BIRTH_DATE: timestamp_ms_to_date(
+                self._account_info.get("birthDate")),
+            ATTR_ACC_OWNER_EMAIL: self._account_info.get("email"),
+            ATTR_ACC_OWNER_PHONE_NUMBER:
+                self._account_info.get("mobilePhoneNumber"),
+        }
+
+        for limit in self._limits:
+            limit_attr_name = ("limit_%s" % limit["limit"]).lower()
+            attributes[limit_attr_name] = limit["amount"]
+
+        return attributes
+
+    @property
+    def icon(self) -> str:
+        """Set the icon for the sensor."""
+        return ICON_ACCOUNT
+
+
+class N26Card(Entity):
+    """Sensor for a N26 card."""
+
+    def __init__(self, client, _account_name, card) -> None:
+        """Initialize a N26 card."""
+        self._client = client
+        self._account_name = _account_name
+        self._card = card
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True.
+
+        Data needs to be polled from the bank servers.
+        """
+        return True
+
+    def update(self) -> None:
+        """Get the current balance and currency for the account."""
+        for card in self._client.get_cards():
```

Comment:
Maybe you want to delegate this update to component level. Just imagine, if you have 5 cards, you have to fetch 5 times 5 cards information, it looks not good.

Reply:
Yes I had the same thought, but I don't quite know how to accomplish this.
How do I trigger a method in the component from a sensor?
I guess from the component I can store the API call result in the `DATA` field just like I already do on the initial component setup. But how does the (already initialised) sensor access this updated information then?

Reply:
You can access hass.data['n26'][CARDS], you just need add one more value in hass.data['n26'] likes last_update_at, if the last_update_at just moment before, you can simply return the cached value. And also need consider it is not thread-safe

Reply:
But how would I ensure thread safety? I can't use simple locks without breaking asyncio can I?

Reply:
No need to overthink. Since everything in your component is using "sync", `update()` call will be sequentially by default.

Reply:
I've added some code to do this. Please have a look :)